### PR TITLE
Documentation/11

### PR DESCRIPTION
**Pull request by kjerandp**
- Fixed some missing types. missing comments, incorrect interfaces etc.
- Regenerated docs
- Updated storybooks ant story names (removed the with prefix)

### CODE REVIEW

File: src/plots/interfaces.ts [hunk 1]
```diff
@@ -1,55 +1,179 @@
...
 
+/**
+ * Data format used by plots
+ * @example [[0, 10], [10, 21], [20, 19] ... ]
+ */
 export type PlotData = Tuplet<number>[];
 
+/**
+ * Data format used by differential plot
+ * @example [[0, 10. 0.5], [10, 21, 0.78], [20, 19, 1.21] ... ]
```
Comment (Brynjulf) [inline src/plots/interfaces.ts]: .
```suggestion
 * @example [[0, 10, 0.5], [10, 21, 0.78], [20, 19, 1.21] ... ]
```

File: src/plots/interfaces.ts [hunk 2]
```diff
@@ -1,55 +1,179 @@
...
+   * Flag to hide/show a plot
+   */
   hidden?: boolean,
+  /**
+   * Set plot orientation. Default is true (horizontally).
+   */
   horizontal?: boolean;
-  data?: DataAccessorFunction, // depricated - use dataAccessor
+  /**
+   * @depricated
```
Comment (Brynjulf) [inline src/plots/interfaces.ts]: .
```suggestion
   * @deprecated
```

File: src/plots/interfaces.ts [hunk 3]
```diff
@@ -1,55 +1,179 @@
...
+  /**
+   * Y axis (range) for plot data
+   */
   domain?: Domain,
+  /**
+   * Set condition for what data to plot
+   */
   defined?: DefinedFunction,
+  /**
+   * Flag to hide/show a plot
```
Comment (Brynjulf) [inline src/plots/interfaces.ts]: what is default flag?

File: src/plots/interfaces.ts [hunk 4]
```diff
@@ -1,55 +1,179 @@
...
+   * offset position to draw the plot
+   * @example 0.5 // renders the plot 50 % off from origin
+   */
   offset?: number,
+  /**
+   * Scale type: 'linear' or 'log'
+   */
   scale?: string,
+  /**
+   * Y axis (range) for plot data
```
Comment (Brynjulf) [inline src/plots/interfaces.ts]: is the domain still only for y axis? could use example/default values

File: src/plots/interfaces.ts [hunk 5]
```diff
@@ -1,55 +1,179 @@
...
+/**
+ * Available differential plot serie options
+ */
 interface DifferentialPlotSerieOptions {
+  /**
+   * Scale type: 'linear' or 'log'
+   */
   scale?: string,
+  /**
+   * Y axis (range) for plot data
```
Comment (Brynjulf) [inline src/plots/interfaces.ts]: is the domain still only for y axis?

File: src/readme.md [hunk 6]
```diff
@@ -5,8 +5,8 @@ Videx Wellog is a set of components developed by the Videx team in Equinor for v
 The library are composed by multiple components that are designed to work together, as independant components and/or as extension points. All relevant classes are exported.
 
 Full-featured components exists in the `./ui` folder, and are:
-* `BasicTrackViewer` - minimalistic container for working with tracks.
-* `WellogComponent` - full-featured container.
+* `TrackGroup` - container class for displaying tracks and handeling user interactions, track titles and legends.
```
Comment (Brynjulf) [inline src/readme.md]: .
```suggestion
* `TrackGroup` - container class for displaying tracks and handling user interactions, track titles and legends.
```

File: src/tracks/graph/interfaces.ts [hunk 7]
```diff
@@ -5,23 +5,63 @@ import { Plot } from '../../plots';
...
+   */
   plots?: PlotConfig[],
+  /**
+   * Data for all plots in the track. May be of any type or a function or promise
+   * returning data. The plots will need to have a data accessor function defined, that
+   * can pick the data it needs from this value.
+   */
   data?: Promise<any> | Function | any,
+  /**
+   * Whether or not to show the loader or not (if configured)
```
Comment (Brynjulf) [inline src/tracks/graph/interfaces.ts]: .
```suggestion
   * Whether or not to show the loader, default: false
```

File: src/tracks/graph/interfaces.ts [hunk 8]
```diff
@@ -5,23 +5,63 @@ import { Plot } from '../../plots';
...
+   * own scale and domain set. Can be 'linear' or 'log'.
+   */
   scale?: string,
+  /**
+   * Domain to use for the graph data range
+   */
   domain?: Domain,
+  /**
+   * Determines if the legend should be clickable to toggle plots on/off when used within a
+   * TrackGroup with graphLegendConfig enabled.
```
Comment (Brynjulf) [inline src/tracks/graph/interfaces.ts]: i assume default is false
```suggestion
   * TrackGroup with graphLegendConfig enabled. Default: false
```

File: src/ui/basic-track-viewer.ts [hunk 9]
```diff
@@ -23,8 +24,8 @@ export default class BasicTrackViewer {
   currentTransform: () => any;
   /**
    * Create instance
-   * @param {Track[]} tracks tracks to be added
-   * @param {number[]} domain y-scale domain
+   * @param tracks tracks to be added
+   * @param domain y-scale domain
```
Comment (Brynjulf) [inline src/ui/basic-track-viewer.ts]: is it still just a domain for the y-scale?

File: src/plots/interfaces.ts (hunk 3 above)
Comment (kjerandp) [reply src/plots/interfaces.ts]: Added default value to comments

File: src/plots/interfaces.ts (hunk 4 above)
Comment (kjerandp) [reply src/plots/interfaces.ts]: Will fix in #24

File: src/plots/interfaces.ts (hunk 5 above)
Comment (kjerandp) [reply src/plots/interfaces.ts]: Will fix in #24

File: src/tracks/graph/interfaces.ts (hunk 8 above)
Comment (kjerandp) [reply src/tracks/graph/interfaces.ts]: Added default, which is true

File: src/ui/basic-track-viewer.ts (hunk 9 above)
Comment (kjerandp) [reply src/ui/basic-track-viewer.ts]: Will fix in #24

**Pull request merged**